### PR TITLE
feat(pwa/cache): clear when disabled

### PR DESCRIPTION
**Pull request by Francois-Esquire**
Clears the Cache Storage when the service worker is disabled or in recovery mode.

## Description

The changes proposed happen mainly on the client in `src/client/service-worker/index.js`. They are run when the service worker is not enabled.

## Motivation and Context

When the service worker is activated, it currently initializes the cache with the `manifest.webmanifest` and the html shell. As we add module caching, we will have other active caches populated. The changes proposed clear the cache when the service worker is not active or in recovery mode.

## How Has This Been Tested?

Testing suite.

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update

## Checklist:

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] Performance tests should be ran against the server prior to merging.
- [x] This change impacts caching for client browsers.
- [ ] This change impacts HTTP headers.
- [ ] This change adds additional environment variable requirements for One App users.
- [ ] I have added the Apache 2.0 license header to any new files created.

## What is the Impact to Developers Using One App?

Removes any cache entries that pertain to One App.

### CODE REVIEW

File: __tests__/integration/one-app.spec.js [hunk 1]
```diff
@@ -978,6 +985,13 @@ describe('Tests that require Docker setup', () => {
           });
 
           expect(result).toBe(null);
+
```
Comment (infoxicator) [inline __tests__/integration/one-app.spec.js:988]: is it worth to add a tests that deletes only the keys cached by the service worker and nothing else, as I understand it, it should keep keys in the cache that don't have anything to do with our SW

File: __tests__/client/service-worker/index.spec.js [hunk 2]
```diff
@@ -84,6 +90,22 @@ describe('initializeServiceWorker', () => {
...
+
+    expect(caches.snapshot()).toEqual({
+      'one-app-test': {
+        'http://localhost/index.js': null,
+      },
+    });
+    await expect(initializeServiceWorker({ serviceWorker: false })).resolves.toBe(registration);
+    expect(registration.unregister).toHaveBeenCalledTimes(1);
+    expect(serviceWorkerClient).not.toHaveBeenCalled();
+    expect(caches.snapshot()).toEqual({});
```
Comment (infoxicator) [inline __tests__/client/service-worker/index.spec.js:106]: is it worth to add a tests that deletes only the keys cached by the service worker and nothing else, as I understand it, it should keep keys in the cache that don't have anything to do with our SW

Comment (Francois-Esquire) [reply __tests__/client/service-worker/index.spec.js:106]: That's a good point. I'll update to add this.

Comment (Francois-Esquire) [reply __tests__/client/service-worker/index.spec.js:106]: I have updated to filter out cache keys created by the service worker with the prefix `__sw` ( 2472120 )

**Pull request merged**